acquia/acquia_cms
# ACMS-970: Added code to programmatically create facet.

**Motivation**
* The search module requires the Article content type to be enabled and article enables other content types. The search module should be enabled in isolation.
Fixes #NNN

**Proposed changes**
* Created the search_category facet programmatically because facet has the dependency on Content types. So the facet gets created only when the content type modules are enabled (the content types are created).

**Alternatives considered**
* None

**Testing steps**
* Scenario 1 (Custom Profile and Enabling the module in isolation)-
** Tested the module in isolation (Using a custom profile and enabling only the search module)
** Visit the /search page once the module is enabled and notice that the /search page opens without any errors - If there is not content created in the system the search page would show up empty.
** Enable any content type module example "acquia_cms_person" and notice that the facet `search_category` is created at page - /admin/config/search/facets/search_category/edit (login as admin user)

* Scenario 2 (Acquia CMS installation)-
** Tested the working of the module in acquia_cms.
** Run simple acquia_cms installation with SS keys and enable the `acquia_cms_starter` module.
** Visit the /search page and notice the search results and all the facets are working correctly.

**Merge requirements**
